pkg/controllers/tokentimeoutsync/tokentimeoutsync.go

```diff
@@ -0,0 +1,158 @@
+
+func NewTokenTimeoutSyncController(
+	accessTokenClient oauthv1client.OAuthAccessTokenInterface,
+	oauthInformers oauthinformer.SharedInformerFactory,
+	configInformers configinformer.SharedInformerFactory,
+	operatorClient v1helpers.OperatorClient,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &tokenTimeoutSyncController{
+		accessTokens:      accessTokenClient,
+		oauthConfigLister: configInformers.Config().V1().OAuths().Lister(),
+		oauthClientLister: oauthInformers.Oauth().V1().OAuthClients().Lister(),
+		operatorClient:    operatorClient, // FIXME: add operator lister to handle unmanaged state
+
+		globalTimeout:  -1,
+		clientTimeouts: map[string]*time.Duration{},
+	}
+
+	return factory.New().ResyncEvery(30*time.Second).WithInformers(
+		oauthInformers.Oauth().V1().OAuthClients().Informer(),
+		configInformers.Config().V1().OAuths().Informer(),
+	).WithSync(c.sync).ToController("TokenTimeoutSyncController", recorder.WithComponentSuffix("wellknown-ready-controller"))
+}
+
+func (c *tokenTimeoutSyncController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	var globalChanged bool
+
+	oauthConfig, err := c.oauthConfigLister.Get("cluster")
+	if err != nil {
+		return err //set condition instead
+	}
+
+	if oauthConfig.Spec.TokenConfig.AccessTokenInactivityTimeout != nil {
+		if newTimeout := oauthConfig.Spec.TokenConfig.AccessTokenInactivityTimeout.Duration; c.globalTimeout != newTimeout { // don't do anything on first run
+			// only update if this is not the first run and the new new timeout is lower than the previous
+			globalChanged = c.globalTimeout != -1 && ((c.globalTimeout > 0 && newTimeout < c.globalTimeout) || (c.globalTimeout == 0 && newTimeout > 0))
+			c.globalTimeout = newTimeout
+		}
+	} else {
+		globalChanged = c.globalTimeout != 0
+		c.globalTimeout = 0
+	}
+
+	tokenList, err := c.accessTokens.List(ctx, metav1.ListOptions{LabelSelector: labels.Everything().String()})
+	if err != nil {
+		return err //set condition instead
+	}
+
+	for _, t := range tokenList.Items {
+		if err := c.updateToken(ctx, &t, globalChanged); err != nil {
+			klog.Errorf("failed to upate token: %v", err) // unsafe - logs token
+		}
+	}
+
+	oauthClients, err := c.oauthClientLister.List(labels.Everything())
+	if err != nil {
+		return err //set condition instead
+	}
+
+	for _, client := range oauthClients {
+		var timeout *time.Duration
+		if client.AccessTokenInactivityTimeoutSeconds == nil {
+			timeout = nil
+		} else {
+			t := time.Duration(*client.AccessTokenInactivityTimeoutSeconds) * time.Second
+			timeout = &t
+		}
+		c.clientTimeouts[client.Name] = timeout
+	}
+
+	return nil
+}
+
+func (c *tokenTimeoutSyncController) updateToken(ctx context.Context, token *oauthv1.OAuthAccessToken, globalChanged bool) error {
+	tokenClientName := token.ClientName
+	if len(tokenClientName) == 0 {
+		return nil
+	}
+
+	tokenClient, err := c.oauthClientLister.Get(tokenClientName)
+	if err != nil {
+		return err
```

Comment:
same as above. any specific not to update it?

Reply:
you don't know what to update it to in this case, do you?